### PR TITLE
Settings.py: Use all instead of default section

### PR DESCRIPTION
**Pull request by hemangsk**
Commit authored by Lasse Schuirmann (sils)

Closes https://github.com/coala/coala-quickstart/issues/127

(Continues https://github.com/coala/coala-quickstart/pull/142)

### CODE REVIEW

File: coala_quickstart/generation/Settings.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def generate_ignore_field(project_dir, languages, extset, ignore_globs):
     for glob in ignore_globs:
         gitignore_files = {file
                            for file in collect_files([glob], null_printer)}
-        if not all_files.isdisjoint(gitignore_files):
+        if all_files.isdisjoint(gitignore_files):
```
Comment (satwikkansal) [inline coala_quickstart/generation/Settings.py]: why did you remove `not`? If set of `all_files` is entirely different from set of `gitignore_files`, we shouldn't add ignore field.

File: tests/generation/SettingsTest.py [hunk 2]
```diff
@@ -65,3 +69,18 @@ def test_allow_complete_section_mode(self):
...
+        ignore_globs = ['main.css', 'file.html', '/repo/hello.html']
+
+        extset = get_extensions(project_files)
+
+        lang_files = split_by_language(project_files)
+
+        res = generate_ignore_field(
+            project_dir, lang_files.keys(), extset, ignore_globs)
+
+        self.assertIn('main.css', res)
```
Comment (satwikkansal) [inline tests/generation/SettingsTest.py]: Imo, we should add another test case where the `ignore_globs` set is disjoint from `project_files` set and make sure the outcome is an empty list.

**Comment by satwikkansal** [comment]
Also, according to https://docs.codecov.io/docs/codecov-delta you'll need to have at least 92.42% coverage on the lines changed in the PR :sweat_smile:

File: coala_quickstart/generation/Settings.py (hunk 1 above)
Comment (hemangsk) [inline coala_quickstart/generation/Settings.py]: Yeah, that's really confusing to me as well. But won't all_files will always be disjoint from gitignore_files due to this https://github.com/hemangsk/coala-quickstart/blob/614598959e007d659454025cc22ccf03f4b85714/coala_quickstart/generation/Settings.py#L58

**Comment by hemangsk** [comment]
Some hidden issue I think, does quickstart actually creates an ```ignore``` field? The code is there, but tests are not. So, I am trying to figure out how to reach that [line](https://github.com/hemangsk/coala-quickstart/blob/614598959e007d659454025cc22ccf03f4b85714/coala_quickstart/generation/Settings.py#L115) of code, to get coverage 93%

File: coala_quickstart/generation/Settings.py (hunk 1 above)
Comment (satwikkansal) [inline coala_quickstart/generation/Settings.py]: hmm, `all_files` should already be ignoring the `ignore_globs` but I don't think it's happening currently (seems like a bug `collect_files` function). 

Also, I thought over the logic and we shouldn't really have this disjoint logic because "If there are no file in `all_files` which is ignored in `.gitignore`, then the `gitignore_files` should be empty because no such files exist"

Comment (satwikkansal) [inline coala_quickstart/generation/Settings.py]: Can't we just do 
```py
ignores = []
for glob in ignore_globs:
    ignores.append(os.path.relpath(glob, project_dir))
return ", ".join(ignores)
```

Comment (satwikkansal) [inline coala_quickstart/generation/Settings.py]: or alternatively, just remove the ignore field altogether

```py
all_files = set(collect_files(
        "**",
        null_printer,
        ignored_file_paths=ignore_globs))
```
and `all_files` now should have all the ignore_files already removed (expectedly) and we can use it to set the `files` setting value for the default section.

Comment (satwikkansal) [inline coala_quickstart/generation/Settings.py]: Not sure which way is better though the first one seems cleaner to me.

**Comment by satwikkansal** [comment]
> Does quickstart actually creates an ignore field?

I tried through different ways, but it's not generating the `ignore` field there's some fault in the logic I guess.

**Comment by satwikkansal** [comment]
Also, can you just check if the following warnings are gone after your changes :sweat_smile: 

image

**Comment by hemangsk** [comment]
@satwikkansal I'll check the warning thingy and will let you know asap

**Comment by satwikkansal** [comment]
Fixed with https://github.com/coala/coala-quickstart/pull/176, closing this one.
